staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/filter.go

```diff
@@ -70,9 +70,6 @@ func (a *evaluationActivation) Parent() interpreter.Activation {
 
 // Compile compiles the cel expressions defined in the ExpressionAccessors into a Filter
 func (c *filterCompiler) Compile(expressionAccessors []ExpressionAccessor, hasParam bool) Filter {
-	if len(expressionAccessors) == 0 {
-		return nil
-	}
```

Comment:
Why to remove this nil check?

Reply:
Now that we have both `validations` and `auditAnnotations`, it is possible for one of the calls to Compile to have an empty slice. We could write special case cost to avoid the compile call, but it is less error prone to make the compile calls safe by not returning an error here.

Reply:
Here is not really a err handling but rather a short circus. I am ok with either since it is nit and not a blocker any way :)